src/Symfony/Component/Ldap/Adapter/ExtLdap/ResultIterator.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Adapter\ExtLdap;
+
+use Symfony\Component\Ldap\Entry;
+
+/**
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class ResultIterator implements \Iterator
+{
+    private $connection;
+    private $search;
+    private $current;
+    private $key;
+
+    public function __construct(Connection $connection, Query $search)
+    {
+        $this->connection = $connection->getResource();
+        $this->search = $search->getResource();
+    }
+
+    /**
+     * Fetches the current entry.
+     *
+     * @return Entry
+     */
+    public function current()
+    {
+        $attributes = ldap_get_attributes($this->connection, $this->current);
+        $dn = ldap_get_dn($this->connection, $this->current);
+
+        return new Entry($dn, $attributes);
+    }
+
+    /**
+     * Sets the cursor to the next entry.
```

Comment:
True. However, this would mean that IDEs would not be able to determine the type of each element. So we need at least to document the `current()` method. Other methods are not necessary, though.